### PR TITLE
CM fixes

### LINKED ISSUE #1747: CM: Client Dom xss
More Info:
https://cx.wdf.sap.corp/cxwebclient/ViewerMain.aspx?scanId=6929148&ProjectID=66426

How to test please contact me.

### PR DESCRIPTION
**Pull request by JohannesDoberer**
Fixes #1747

### CODE REVIEW

File: website/fiddle/public/auth/idpmock/logout.html [hunk 1]
```diff
@@ -2,27 +2,27 @@
...
+  Logging out...
+  <script>
+    window.onload = function () {
+      let href = encodeURI(window.location.href);
+      const decodeChars = (string) => {
+        return string.replace(/[<>'*+?^${}()|[\]\\]/g, '\\$&')
+      }
+      let redirectTo =
+        decodeChars(decodeURIComponent(href.match(/post_logout_redirect_uri=(.*?)(&|$)/)[1]));
+      // redirectTo = decodeChars(redirectTo);
```
Comment (UlianaMunich) [inline website/fiddle/public/auth/idpmock/logout.html:18]: cleanup?

**Review by stanleychh — APPROVED** [review]
Besides the cleanup comment, the rest looks good to me.

File: website/fiddle/public/auth/idpmock/logout.html [hunk 2]
```diff
@@ -2,27 +2,27 @@
...
-    window.location.href = redirectTo; 
-  };
-	</script>
+  Logging out...
+  <script>
+    window.onload = function () {
+      let href = encodeURI(window.location.href);
+      const decodeChars = (string) => {
+        return string.replace(/[<>'*+?^${}()|[\]\\]/g, '\\$&')
+      }
```
Comment (stanleychh) [inline website/fiddle/public/auth/idpmock/logout.html:15]:
```suggestion
      const decodeChars = (string) => string.replace(/[<>'*+?^${}()|[\]\\]/g, '\\$&');
```

**Review by UlianaMunich — APPROVED** [review]
Tested, works well!

**Pull request merged**